### ISSUE TITLE
Masking in sequential model

### ISSUE THREAD

**Issue by ehsanasgari** [fchollet/keras#3390]
Hi there, 

I am working on a seq 2 seq regression (every embedding in the input sequence goes to a real value). Since my input length varies I am using masking. I was wondering if masking needs to be done only at the input layer or do I need to have masking at the output as well? Here is my model specification:

```
    # Model definition         
    self.model = Sequential()
    # to remove items from the shorter sequences than the max length
    self.model.add(Masking(5.000000, input_shape=(max_length, self.embedding_mat.shape[1])))
    self.model.add(LSTM(hidden_size, return_sequences=True))
    self.model.add(TimeDistributed(Dense(1,activation="relu")))
    self.model.compile(loss='mse', optimizer='adadelta',metrics=["mse"])
```

Thank you for your help.

**Comment by dragonhyq** [comment]
#957
